main.cpp

```diff
@@ -39,64 +43,99 @@ QGamepadManager::GamepadButtons buttons = 0;
 QPoint touchScreenPosition;
 
 QSettings settings("TuxSH", "InputRedirectionClient-Qt");
 
+QGamepadManager::GamepadButton getButton(QString dsBtn)
+{
+    QString gpPadBtn;
+    QVariant val;
+    QSettings settings("GamepadControls.ini", QSettings::IniFormat);
+
+    val =  settings.value(dsBtn);
+    gpPadBtn = val.toString();
+
+    QGamepadManager::GamepadButton btnConv;
+
+    if(gpPadBtn == "360_A") btnConv = QGamepadManager::ButtonA;
+    else
+    if(gpPadBtn == "360_B") btnConv = QGamepadManager::ButtonB;
+    else
+    if(gpPadBtn == "360_X") btnConv = QGamepadManager::ButtonX;
+    else
+    if(gpPadBtn == "360_Y") btnConv = QGamepadManager::ButtonY;
+    else
+    if(gpPadBtn == "360_BACK") btnConv = QGamepadManager::ButtonSelect;
+    else
+    if(gpPadBtn == "360_START") btnConv =  QGamepadManager::ButtonStart;
+    else
+    if(gpPadBtn == "360_D_UP") btnConv =  QGamepadManager::ButtonUp;
+    else
+    if(gpPadBtn == "360_D_DOWN") btnConv =  QGamepadManager::ButtonDown;
+    else
+    if(gpPadBtn == "360_D_LEFT") btnConv = QGamepadManager::ButtonLeft;
+    else
+    if(gpPadBtn == "360_D_RIGHT") btnConv = QGamepadManager::ButtonRight;
+    else
+    if(gpPadBtn == "360_R1") btnConv = QGamepadManager::ButtonR1;
+    else
+    if(gpPadBtn == "360_R2") btnConv = QGamepadManager::ButtonR2;
+    else
+    if(gpPadBtn == "360_R3") btnConv = QGamepadManager::ButtonR3;
+    else
+    if(gpPadBtn == "360_L1") btnConv = QGamepadManager::ButtonL1;
+    else
+    if(gpPadBtn == "360_L2") btnConv = QGamepadManager::ButtonL2;
+    else
+    if(gpPadBtn == "360_L3") btnConv = QGamepadManager::ButtonL3;
+    else
+    btnConv = QGamepadManager::ButtonGuide;
+
+    return btnConv;
+}
+
 void sendFrame(void)
 {
     static const QGamepadManager::GamepadButton hidButtonsAB[] = {
-        QGamepadManager::ButtonA,
-        QGamepadManager::ButtonB,
+        getButton("Button/3DS_A"), //3ds_A
+        getButton("Button/3DS_B"), //3ds_B
     };
 
     static const QGamepadManager::GamepadButton hidButtonsMiddle[] = {
-        QGamepadManager::ButtonSelect,
-        QGamepadManager::ButtonStart,
-        QGamepadManager::ButtonRight,
-        QGamepadManager::ButtonLeft,
-        QGamepadManager::ButtonUp,
-        QGamepadManager::ButtonDown,
-        QGamepadManager::ButtonR1,
-        QGamepadManager::ButtonL1,
+        getButton("Button/3DS_SELECT"),
+        getButton("Button/3DS_START"),
+        getButton("Button/3DS_D_RIGHT"),
+        getButton("Button/3DS_D_LEFT"),
+        getButton("Button/3DS_D_UP"),
+        getButton("Button/3DS_D_DOWN"),
+        getButton("Button/3DS_R"),
+        getButton("Button/3DS_L"),
     };
 
     static const QGamepadManager::GamepadButton hidButtonsXY[] = {
-        QGamepadManager::ButtonX,
-        QGamepadManager::ButtonY,
+        getButton("Button/3DS_X"),
+        getButton("Button/3DS_Y"),
     };
 
     static const QGamepadManager::GamepadButton irButtons[] = {
-        QGamepadManager::ButtonR2,
```

Comment:
So I just happened to come across a new 3ds (convenient huh?) and have updated with ZL and ZR support.